application/src/main/java/org/togetherjava/tjbot/commands/help/BookmarksHelper.java

```diff
@@ -0,0 +1,109 @@
+
+/**
+ * This is a utility class for interacting with the database
+ */
+public class BookmarksHelper {
+
+    private BookmarksHelper() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    protected static Optional<HelpThreadsRecord> getHelpThread(Database database, long channelId) {
+        return database.read(context -> context.selectFrom(HELP_THREADS)
+            .where(HELP_THREADS.CHANNEL_ID.eq(channelId))
+            .fetch()
+            .stream()
+            .findFirst());
+    }
+
+    protected static Optional<BookmarksRecord> getBookmark(Database database, long creatorId,
+            long channelId) {
+        return database.read(context -> context.selectFrom(BOOKMARKS)
+            .where(BOOKMARKS.CREATOR_ID.eq(creatorId), BOOKMARKS.CHANNEL_ID.eq(channelId))
+            .stream()
+            .findFirst());
+    }
+
+    protected static void addBookmark(Database database, long creatorId, long channelId,
+            String originalTitle, @Nullable String note) {
+        database.write(context -> context.newRecord(BOOKMARKS)
+            .setCreatorId(creatorId)
+            .setChannelId(channelId)
+            .setOriginalTitle(originalTitle)
+            .setLastRenewedAt(Instant.now())
+            .setLastActivityAt(Instant.now())
+            .setNote(note)
+            .insert());
+    }
+
+    protected static void removeBookmark(Database database, long creatorId, long channelId) {
+        database.write(context -> context.deleteFrom(BOOKMARKS)
+            .where(BOOKMARKS.CREATOR_ID.eq(creatorId), BOOKMARKS.CHANNEL_ID.eq(channelId))
+            .execute());
+    }
+
+    protected static List<BookmarksRecord> getUserBookmarks(Database database, long creatorId) {
+        return database.read(context -> context.selectFrom(BOOKMARKS)
+            .where(BOOKMARKS.CREATOR_ID.eq(creatorId))
+            .orderBy(BOOKMARKS.LAST_ACTIVITY_AT.desc())
+            .stream()
+            .toList());
+    }
+
+    protected static void renewBookmark(Database database, long creatorId, long channelId) {
+        database.write(context -> context.update(BOOKMARKS)
+            .set(BOOKMARKS.LAST_RENEWED_AT, Instant.now())
+            .where(BOOKMARKS.CREATOR_ID.eq(creatorId), BOOKMARKS.CHANNEL_ID.eq(channelId))
+            .execute());
+    }
+
+    protected static void updateActivity(Database database, long channelId) {
+        database.write(context -> context.update(BOOKMARKS)
+            .set(BOOKMARKS.LAST_ACTIVITY_AT, Instant.now())
+            .where(BOOKMARKS.CHANNEL_ID.eq(channelId))
+            .execute());
+    }
+
+    protected static void cleanupUserBookmarks(Database database, long creatorId) {
+        database.write(context -> context.deleteFrom(BOOKMARKS)
+            .where(BOOKMARKS.CREATOR_ID.eq(creatorId),
+                    BOOKMARKS.LAST_RENEWED_AT.lessThan(Instant.now()
+                        .minus(BookmarksCommand.EXPIRE_DELAY_SECONDS, ChronoUnit.SECONDS)))
```

Comment:
this is a bit complex to read. id put that instant which ur constructing into a separate variable with a nice name